t/lib/t/MusicBrainz/Server/Data/Editor.pm

```diff
@@ -174,11 +178,39 @@ test 'find_by_email and is_email_used_elsewhere' => sub {
     );
     ok(
         $editor_data->is_email_used_elsewhere(
-            'EDITOR@example.com',
+            'EDITOR@EXAMPLE.COM',
             $new_editor_2->id + 1,
         ),
         'The email is shown to be in use even if searching with all caps',
     );
+
+    note('We set an all caps email for the new editor with update_email');
+    $editor_data->update_email($new_editor_2, 'EDITOR@EXAMPLE.COM');
+
+    note('We search for the new editor email with find_by_email');
+    my @editors = $editor_data->find_by_email('EDITOR@EXAMPLE.COM');
+    is(scalar(@editors), 1, 'An editor was found with the exact email');
+    is($editors[0]->id, $new_editor_2->id, 'The right editor was found');
+
+    @editors = $editor_data->find_by_email('editor@example.com');
+    is(scalar(@editors), 1, 'An editor was found searching with normal caps');
+    is($editors[0]->id, $new_editor_2->id, 'The right editor was found');
+
+    note('We check is_email_used_elsewhere shows the email as being in use');
+    ok(
+        $editor_data->is_email_used_elsewhere(
+            'EDITOR@EXAMPLE.COM',
+            $new_editor_2->id + 1,
```

Comment:
Not sure where `$new_editor_2->id + 1` is coming from. Or is that supposed to just represent something like `$nonexistent_id`?

Reply:
Yeah, it's just meant to be "someone else" since `is_email_used_elsewhere` needs an editor ID that wants to use the email. Maybe I just should make it a `$someone_else_id`.